sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -176,6 +180,63 @@ class weights.
         """
         return sample_weight
 
+    def _check_X(self, X, *, reset):
```

Comment:
A docstring with a list of attributes that are set would not hurt.
Also, should we adjust the function name?